aim/ext/notebook/notebook.py

```diff
@@ -106,16 +115,21 @@ def display_colab(port, display):
         IPython.display.display(script)
 
 
-def display_notebook(host, port, display):
+def display_notebook(host, port, display, proxy_url=None):
     """Display Aim instance in an ipython context output frame.
     """
     import IPython.display
-    shell = """
-          <iframe id="aim" width="100%" height="800" frameborder="0" src={}:{}{}>
-          </iframe>
-        """.format(host, port, '/notebook/')
+    url = "{}:{}{}".format(host, port, _NOTEBOOK_PATH_POSTFIX)
 
-    # @TODO write passing proxy logic
+    if proxy_url:
```

Comment:
What happens if user does not provide `proxy_url`? Can we auto-detect SageMaker or notebook behind firewall and ask user to provide one?

Reply:
Yeah! I couldn' find a stable way to detect sagemaker environment fastly, there is no environment variable which will tell us the what environment is here.
But we can check the root environment name, has `/SageMaker` folder or not?, or `jupyter-server-proxy` is installed or not. Will choose a way which will cover 90% of usage, because there is no a way to detect it for all cases.

Lets do a stable release without it, and add enhancement with patch releases of 3.6.x.